rfcs/0145-doc-strings.md

```diff
@@ -0,0 +1,504 @@
+
+In general, we needed the following:
+
+1. General format for doc-comments.
+2. Format for headings and the allowed content.
+
+> It would be nice if this could be close to the markdown format.
+> Markdown is the most straightforward and most accepted format
+> for writing and rendering documentation.
+
+## General Formats
+
+| Property / Approach | `##` | `/** */` | `Javadoc` |
+|---|---|---|---|
+| Inspired by | Rust | Current nixpkgs.lib | C++/Java/Javascript |
+| Changes the existing code by | Much | Less | Even More |
+| Needs Termination | No | Yes | Yes |
+| Indentation | Clear | Poor | Poor |
+| Needs vertical space  | No | Yes | Yes |
+| Visual distinction from comments | High | Low | Medium |
+| Needs Autocompletion (Language Server) to continue the next line. | Yes | No | Yes |
+| Punctuation Variations / Amount of different special characters | Less | More | More |
+| Markdown compatibility (also depends on indentation clarity) | Good, but visual conflicts with headings` #` | Poor | Medium |
+| breaks when interrupted with newlines | Yes | No | ? |
+
+**Proposed format:**
+
+Use `##` to start a doc-comment. This allows clear visual separation from regular comments.
+And provides a good compatibility with the strived markdown content.
+
+### Refactoring note
+
+nixpkgs comments:
+
+- `##` ~4k usages (most of them are used for visual separation e.g. `###########`)
+- `#` ~20k usages
+- `/*` ~6k usages
+- `/**` 160 usages (most of them are completely empty ?)
+
+## General Headings
+
+| Property / Approach | `# <Heading>` | `@<Heading>:` |
+|---|---|---|
+| Inspired by | Markdown | Doxygen |
+| Changes the existing code by | Minor | Minor |
+| Needs vertical space  | Recommended | No |
+| Visual distinction from comments | Low | High |
+| Markdown compatibility | Best | None |
+
+**Proposed headings:**
+
+Use markdown headings `# <Heading>`. This allows best compatibility with the already specified markdown/commonmark format. Allowing for easy and intuitive usage for comments.
+
+## Consequences to not implementing this
+
+- By not implementing this feature, nix gains no ability for tool-generated documentation.
+- Documentation will be defined by nixdoc, not by the nix community itself.
+- Many existing comments written for documentation will remain un-discoverable.
+
+## Examples
+
+This section contains examples for the different formats to visualize them and emphasize the previously discussed characteristics.
+
+### `##` inspired from rust's `///` 
+
+with `Markdown` Headings
+
+`somefile.nix`
+
+```nix
+    ## <Description or Tagline>
+    ## 
+    ## # Example
+    ##
+    ## <Comprehensive code>
+    ## 
+    ## # Type
+    ##    
+    ## <Type Signature>
+    mapAttrs = f: s: #...
+```
+
+### `/** */` inspired by the current multiline strings 
+
+With `@{keyword}:` Headings
+
+```nix
+    /** 
+        <Description or Tagline>
```

Comment:
I believe it should follow the CommonMark spec, as you suggested on the previous heading.
It makes the things more consistent.